snap/info.go

```diff
@@ -1010,21 +1011,18 @@ func (app *AppInfo) ServiceFile() string {
 	return filepath.Join(dirs.SnapServicesDir, app.ServiceName())
 }
 
-// Env returns the app specific environment overrides
-func (app *AppInfo) Env() []string {
-	appEnv := app.Snap.Environment.Copy()
-	for _, k := range app.Environment.Keys() {
-		appEnv.Set(k, app.Environment.Get(k))
-	}
-
-	return envFromMap(appEnv)
-}
-
 // IsService returns whether app represents a daemon/service.
 func (app *AppInfo) IsService() bool {
 	return app.Daemon != ""
 }
 
+func (app *AppInfo) EnvStack() []osutil.ExpandableEnv {
```

Comment:
this lost the doc comment, also to keep the terminology smaller maybe we should call this EnvChain